### ISSUE TITLE
Feature request: Checksum certain files as "endpoints" of the analysis

### ISSUE THREAD

**Issue by cjvanlissa** [cjvanlissa/worcs#72]
I'm implementing an "entry point" in the .worcs file; if we also allow users to define certain files as end points, then we can store checksums for the files (using both raw and synthetic data), and we could make sure that running the entry point file will result in correctly checksummed end point files.

This would enable an automatic check of reproducibility.

Also relevant for `repro`

**Comment by brandmaier** [comment]
Great idea. It should be possible to setup github actions that runs this consistency check whenever you commit changes to github. I am sure that @aaronpeikert knows how to do this right away.

**Comment by cjvanlissa** [comment]
You mean like using Travis? For `repro`, I think that's an excellent idea - for `worcs`, I'd be satisfied with a check that can be run locally. The steps of connecting Travis to GitHub are too involved for the target audience for `worcs`, I think

**Comment by brandmaier** [comment]
Correct. I don't mind which packages will have the functionality. According to the `repro`-logic, it could be something like an R function `use_travis_for_worcs_consistency_check()` or similar, which would setup a travis file that simply runs your check function with travis. Again, a great way how these approaches could complement each other.

**Comment by cjvanlissa** [comment]
Yup. This should work. We could have:

`worcs::consistency_check()`:
* Run entry point
* Checksum exit point(s)

`repro::use_travis_consistency_check()`
* Do the above, but on Travis using `testthat`

**Comment by aaronpeikert** [comment]
Supporting GitHub actions is definitely on my list (aaronpeikert/repro#35) and should be easy enough, finding a good "architecture" for the entry point/endpoints is more difficult I think.
What makes it so difficult is that I'd like a general system for storing metadata (aaronpeikert/repro#30) that is just complex enough to be useful for a wide range of things we might want to store (we talked about tags for databases etc.) but doesn't reimplement a poor mans version of git (or checking file versions via checksums) or make (for entry and endpoint).
